spec/fastsync/vdd/fastsyncL2.tla

```diff
@@ -0,0 +1,679 @@
+   Returns NilPeer in case there are no such peer.  *)
+FindPeerToServe(bPool, h) == 
+    LET peersThatCanServe == { p \in bPool.peerIds: 
+                /\ bPool.peerHeights[p] >= h 
+                /\ NumOfPendingRequests(bPool, p) < PEER_MAX_REQUESTS } IN
+    
+    LET pendingBlocks == 
+        {i \in Heights: 
+            /\ i >= bPool.height 
+            /\ \/ bPool.pendingBlocks[i] /= NilPeer
+               \/ bPool.blockStore[i] /= NilBlock
+        } IN
+    
+    IF \/ peersThatCanServe = {} 
+       \/ Cardinality(pendingBlocks) >= TARGET_PENDING 
+    THEN NilPeer 
+    \* pick a peer that can serve request for height h that has minimum number of pending requests
+    ELSE CHOOSE p \in peersThatCanServe: \A q \in peersThatCanServe:       
+            /\ NumOfPendingRequests(bPool, p) <= NumOfPendingRequests(bPool, q) 
+            /\ p < q
+    
+       
+\* Make a request for a block (if possible) and returns a request message and block poool.
+CreateRequest(bPool) ==
+    LET nextHeight == FindNextRequestHeight(bPool) IN
+    
+    IF nextHeight = NilHeight THEN [msg |-> NoMsg, pool |-> bPool]
+    ELSE
+     LET peer == FindPeerToServe(bPool, nextHeight) IN
+     IF peer = NilPeer THEN [msg |-> NoMsg, pool |-> bPool]
+     ELSE 
+        LET m == [type |-> "blockRequest", peerId |-> peer, height |-> nextHeight] IN
+        LET newPool == [bPool EXCEPT 
+                          !.pendingBlocks = [bPool.pendingBlocks EXCEPT ![nextHeight] = peer]
+                        ] IN
+        [msg |-> m, pool |-> newPool]   
+        
+
+\* Returns node state, i.e., defines termination condition. 
+GetState(bPool) == 
+    IF bPool.syncedBlocks = 0 \* corresponds to the syncTimeout in case no progress has been made for a period of time.
+    THEN "aborted"
+    ELSE IF /\ bPool.height > 1 
+            /\ bPool.height >= MaxPeerHeight(bPool) \* see https://github.com/tendermint/tendermint/blob/61057a8b0af2beadee106e47c4616b279e83c920/blockchain/v2/scheduler.go#L566
+         THEN "finished" 
+         ELSE "running"      
+         
+(* Verify if commit is for the given block id and if commit has enough voting power. 
+   See https://github.com/tendermint/tendermint/blob/61057a8b0af2beadee106e47c4616b279e83c920/blockchain/v2/processor_context.go#L12 *)
+VerifyCommit(blockId, lastCommit) == 
+    /\ lastCommit.enoughVotingPower     
+    /\ lastCommit.blockId = blockId 
+
+
+(* Tries to execute next block in the pool, i.e., defines block validation logic. 
+   Returns new block pool (peers that has send invalid blocks are removed).  *)
+ExecuteBlocks(bPool) ==  
+    LET bStore == bPool.blockStore IN
+    LET block1 == bStore[bPool.height] IN
+    LET block2 == bStore[bPool.height+1] IN
+    
+    IF block1 = NilBlock \/ block2 = NilBlock \* we don't have two next consequtive blocks
+    THEN bPool   
+    ELSE IF bPool.height > 1 /\ ~VerifyCommit(block1.blockId, block2.lastCommit) 
```

Comment:
?
```suggestion
    ELSE IF bPool.height >= 1 /\ ~VerifyCommit(block1.blockId, block2.lastCommit) 
```
I believe `bPool.height` is initialized with 1, not sure how this advances then.